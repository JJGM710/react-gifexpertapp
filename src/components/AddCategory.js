import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({setCategories, categories}) => {

    const [inputValue,setInputValue]= useState('')

    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setInputValue(e.target.value)


        console.log('llamdo de evento onchange')
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        //console.log('handleSubmit', inputValue)
        let test = categories.find((ctg) => ctg.toLowerCase() === inputValue.trim().toLowerCase());
        if (inputValue.trim().length > 2 && test === undefined)
        {
            setCategories(ctg => [inputValue, ...ctg]);
            console.log('se hizo el cambio')
            setInputValue('');
        }
       // console.log(categories);
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input 
                placeholder="Search.."
                className='box'
                type = "text" 
                value = {inputValue}
                onChange = {handleInputChange}
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired,
    categories: PropTypes.array.isRequired
};

export default AddCategory;
