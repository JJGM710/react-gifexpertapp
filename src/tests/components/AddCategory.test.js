import  React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import AddCategory from './../../components/AddCategory';

describe('Pruebas en AddCategory', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(< AddCategory setCategories={setCategories} categories={['One punch']}/>);
    
    beforeEach(()=>{
        // hace un clean de la funciones mock que hayamos inicializado
        jest.clearAllMocks();
        wrapper = shallow(< AddCategory setCategories={setCategories} categories={['One punch']}/>);

    });
    test('debe renderizar correctamente ', () => {


        expect(wrapper).toMatchSnapshot();
        
    })

    test('debe  cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const testinput = 'Hola Mundo';
        //dispara evento de una caja de texto  aka input
        input.simulate('change', {target : {value : testinput}});
       // console.log(wrapper.find('p').text().trim())
        expect(wrapper.find('p').text().trim()).toBe(testinput);
        
    })
    
    test('no debe de postear la informacion onSubmit', () => {

        wrapper.find('form').simulate('submit', {preventDefault(){}});

        expect(setCategories).not.toHaveBeenCalled();
        
    })
    
    test('debe de llamar el setCategories y limpiar la caja de text', () => {

        //simular el inputChange

        const input = wrapper.find('input');
        const testinput = 'Hola Mundo';
        input.simulate('change', {target : {value : testinput}});
        const test1= wrapper.find('p').text().trim()
        console.log(test1);
        // simular el submit
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        // se debe haber llamado setCategories
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        // el valor del input debe estar como un string vacio, dos  maneras 
        //1era
        const test2alt = wrapper.find('input').props().value;
        console.log(test2alt);
        expect(test2alt).toBe('');
        //2da
        const test2= wrapper.find('p').text().trim()
        expect(test2).toBe('');
        
    })
    
    
    
})
