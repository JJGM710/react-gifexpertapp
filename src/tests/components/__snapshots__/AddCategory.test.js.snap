// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory debe renderizar correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    className="box"
    onChange={[Function]}
    placeholder="Search.."
    type="text"
    value=""
  />
</form>
`;
